snapcraft/parts/lifecycle.py

```diff
@@ -317,7 +344,12 @@ def _run_in_provider(
     provider = providers.get_provider(provider_name)
     provider.ensure_provider_is_available()
 
```

Comment:
`def _run_in_provider()` has no unit tests :(

After trying to write the unit tests this afternoon, I understand why none exist.  Mocking the `Executor` objects is not trivial.  Charmcraft has a complex solution and I'm not good enough with pytest to get it working for snapcraft in an afternoon.